### PR TITLE
Modify search by postal to use PostalCode attribute

### PR DESCRIPTION
**Pull request by yyanyee**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 1]
```diff
@@ -22,13 +22,20 @@ public class FindCommand extends Command {
...
 
+    /**
+     * Constructs a FindCommand with the specified predicate.
+     *
+     * @param predicate The predicate used to filter persons in the address book.
+     * @throws NullPointerException if the predicate is null.
+     */
     public FindCommand(Predicate<Person> predicate) {
         this.predicate = requireNonNull(predicate);
+        System.out.println(this.predicate);
```
Comment (noth-k) [inline src/main/java/seedu/address/logic/commands/FindCommand.java:38]: Was this left in for debugging?

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 2]
```diff
@@ -50,12 +57,12 @@ public boolean equals(Object other) {
         }
 
         FindCommand otherFindCommand = (FindCommand) other;
+        System.out.println("predicate class : " + predicate.getClass());
```
Comment (noth-k) [inline src/main/java/seedu/address/logic/commands/FindCommand.java:60]: Same comment as above.

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 3]
```diff
@@ -51,38 +50,43 @@ public FindCommand parse(String args) throws ParseException {
         Predicate<Person> phonePredicate = new PhoneContainsKeywordsPredicate(phoneKeywords);
         Predicate<Person> postalPredicate = new PostalContainsKeywordsPredicate(postalKeywords);
 
-        if (!nameKeywords.isEmpty() && !phoneKeywords.isEmpty() && !postalKeywords.isEmpty()) { // all 3 searches
+        System.out.println(namePredicate.getClass());
+        System.out.println(phonePredicate.getClass());
+        System.out.println(postalPredicate.getClass());
+
+        if (!nameKeywords.isEmpty() && !phoneKeywords.isEmpty() && !postalKeywords.isEmpty()) {
```
Comment (noth-k) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:57]: This section appears to have alot of branching logic with repeated checks. Could this be possibly refactored?

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 4]
```diff
@@ -33,16 +33,15 @@ public FindCommand parse(String args) throws ParseException {
 
         String[] keywords = trimmedArgs.split("\\s+");
 
-        // Separate keywords into phone and name
         List<String> phoneKeywords = Arrays.stream(keywords)
                 .filter(this::isNumeric)
+                .filter(keyword -> keyword.length() != 6)
```
Comment (noth-k) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:38]: Maybe the isPostalCode() method could be used here to improve readability?

File: src/main/java/seedu/address/model/person/PhoneContainsKeywordsPredicate.java [hunk 5]
```diff
@@ -26,8 +27,9 @@ public PhoneContainsKeywordsPredicate(List<String> keywords) {
      */
     @Override
     public boolean test(Person person) {
-        return keywords.stream()
-                .anyMatch(keyword -> person.getPhone().value.contains(keyword));
+        return keywords.stream().anyMatch(keyword ->
+                person.getPhone().toString().contains(keyword) // Checks if the phone number contains the keyword
+        );
```
Comment (noth-k) [inline src/main/java/seedu/address/model/person/PhoneContainsKeywordsPredicate.java:32]: Maybe the comment can be removed as the java docs seems to explain the code?

File: src/test/java/seedu/address/logic/commands/FindCommandTest.java [hunk 6]
```diff
@@ -185,10 +177,10 @@ private Predicate<Person> preparePredicate(String userInput) {
     }
 
     private boolean isNumeric(String str) {
-        return str.matches("\\d+");
+        return str.matches("\\d+") && str.length() != 6;
```
Comment (noth-k) [inline src/test/java/seedu/address/logic/commands/FindCommandTest.java:180]: This code appears to be repeated. Perhaps you could import this method and use it following the DRY principle?

File: src/test/java/seedu/address/logic/commands/FindCommandTest.java [hunk 7]
```diff
@@ -185,10 +177,10 @@ private Predicate<Person> preparePredicate(String userInput) {
...
     }
-
     private boolean isPostalCode(String str) {
```
Comment (noth-k) [inline src/test/java/seedu/address/logic/commands/FindCommandTest.java:182]: The same comment as the one above.

File: src/test/java/seedu/address/logic/commands/FindCommandTest.java [hunk 8]
```diff
@@ -159,6 +150,7 @@ private Predicate<Person> preparePredicate(String userInput) {
 
         List<String> phoneKeywords = Arrays.stream(keywords)
                 .filter(this::isNumeric)
+                .filter(keyword -> keyword.length() != 6)
```
Comment (noth-k) [inline src/test/java/seedu/address/logic/commands/FindCommandTest.java:153]: Again, perhaps isPostalCode() could be used here?

**Review by noth-k — COMMENTED** [review]
The code looks ok; some code lines seem to be repeated which maybe could be avoided and some print lines were left in but nice job. Perhaps could improve on code coverage soon.

File: src/test/java/seedu/address/model/person/PhoneContainsKeywordsPredicateTest.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.testutil.PersonBuilder;
+
+public class PhoneContainsKeywordsPredicateTest {
+
+    @Test
+    public void equals() {
+        List<String> firstPredicateKeywordList = Collections.singletonList("91234567");
```
[the remark this answers is not in the document]
Comment (yyanyee) [reply src/test/java/seedu/address/model/person/PhoneContainsKeywordsPredicateTest.java:18]: follow format given in `NameContainsKeywordsPredicateTest`

File: src/test/java/seedu/address/model/person/PostalContainsKeywordsPredicateTest.java [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+        assertTrue(predicate.test(new PersonBuilder().withPostalCode("654321").build()));
+
+        // Mixed-case keywords (though postal codes are numeric)
+        predicate = new PostalContainsKeywordsPredicate(Arrays.asList("123456"));
+        assertTrue(predicate.test(new PersonBuilder().withPostalCode("123456").build()));
+    }
+
+    @Test
+    public void test_postalDoesNotContainKeywords_returnsFalse() {
+        // Zero keywords
```
[the remark this answers is not in the document]
Comment (yyanyee) [reply src/test/java/seedu/address/model/person/PostalContainsKeywordsPredicateTest.java:66]: followed format given in `NameContainsKeywordsPredicateTest`

**Comment by yyanyee** [comment]
will improve codecov

**Pull request merged**